### DISCUSSION TITLE
Can you help with an example of "predictAccount"

### DISCUSSION THREAD

**Discussion by mohan-the-octocat** [ananthakumaran/paisa#49] [Q&A] [answered]
👋 @ananthakumaran. 🙏 for the 🏆 tool!

Can you provide some examples of the `predictAccount` functionality? ([here](https://paisa.fyi/reference/import/#predictaccountcorpus-string-prefix-string-string)).

I am unable to understand the references to `corpus` here?

**Comment by ananthakumaran** [comment, marked as answer]
Think of predictAccount as doing a search & ranking the results, so it's going to search across all your existing transactions. Now you need to provide the query to match which is called as corpus in the doc (I think it's not a great name, I will update with a better one soon). The match will be performed against the description of the transaction (and amount)

```
2001-01-01   Salary jan
          ....

2001-01-01   Rent
          ...
````

predictAccount(salary) would likely match the first transaction and use the account from that. Normally, you pass the same value you used in the description

**Comment by mohan-the-octocat** [comment]
🙇 thanks for the quick revert.

Please confirm if I am understanding this correctly.

`predictAccount` is meant to search previous transactions (treats as the `corpus`) and automatically assign the balancing account based on those past occurrences. 

For a past Salary transactions that looks like the below...

```
2023/07/28 - My Salary 
    Assets:Checking:Axis1234                 100 INR
    Income:NEFT/CITIN23413141026/GREAT EMPLOYER/SALARY T        -100 INR
```

For any subsequent Salary transaction imported from the bank statement, `predictAccount` would automatically populate the value of "Checking:Axis1234"
```
2023/08/28 - 
    Assets:Checking:Axis1234                 100 INR
    Income:NEFT/CITIN23413141026/GREAT EMPLOYER/SALARY T        -100 INR
```

Is that correct?!

**Comment by ananthakumaran** [comment]
Yes, correct
